service/src/test/java/cz/muni/fi/service/CarFacadeTest.java

```diff
@@ -0,0 +1,172 @@
+package cz.muni.fi.service;
+
+
+import cz.muni.fi.dao.CarDao;
+import cz.muni.fi.dao.ComponentDao;
+import cz.muni.fi.dao.DriverDao;
+import cz.muni.fi.dto.CarDTO;
+import cz.muni.fi.dto.ComponentDTO;
+import cz.muni.fi.dto.DriverDTO;
+import cz.muni.fi.entities.Car;
+import cz.muni.fi.entities.Component;
+import cz.muni.fi.entities.Driver;
+import cz.muni.fi.facade.CarFacade;
+import cz.muni.fi.persistanceEnums.ComponentType;
+import cz.muni.fi.service.config.ServiceConfiguration;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Richard Hrmo
+ */
+
+@ContextConfiguration(classes = ServiceConfiguration.class)
+public class CarFacadeTest extends AbstractTestNGSpringContextTests {
+
+    @Autowired
+    private CarFacade carFacade;
+
+    @Autowired
+    private CarDao carDao;
+
+    @Autowired
+    private DriverDao driverDao;
+
+    @Autowired
+    private ComponentDao componentDao;
+
+    @Autowired
+    private BeanMappingService beanMappingService;
+
+    private List<CarDTO> carsList = new ArrayList<>();
+    private List<Component> components = new ArrayList<>();
+
+    private CarDTO redbullF1;
+    private CarDTO jordan;
+    private CarDTO ferrari;
+
+    private DriverDTO ricciardo = new DriverDTO();
```

Comment:
Are 4 drivers needed here? I suggest reducing number of drivers and also cleaning setup method bellow